### PR TITLE
EWS: Set most recent period as default filter value

### PR DESCRIPTION
**Pull request by kkrampa**
http://manage.dimagi.com/default.asp?190526#1068184

### CODE REVIEW

File: custom/ewsghana/tests/__init__.py [hunk 1]
```diff
@@ -19,3 +19,4 @@
 from .test_notifications import *
 from .ews_test_reminders import *
 from .handlers.stop import *
+from .test_calculate_last_period import *
```
Comment (dimagimon) [inline custom/ewsghana/tests/__init__.py]: W292 no newline at end of file

File: custom/ewsghana/tests/test_calculate_last_period.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        self.assertEqual(calculate_last_period(monday), expected_period)
+        self.assertEqual(calculate_last_period(tuesday), expected_period)
+        self.assertEqual(calculate_last_period(wednesday), expected_period)
+        self.assertEqual(calculate_last_period(thurdsay), expected_period)
+
+        expected_period = (datetime(2016, 1, 15), datetime(2016, 1, 21, 23, 59, 59, 999999))
+
+        self.assertEqual(calculate_last_period(friday), expected_period)
+        self.assertEqual(calculate_last_period(saturday), expected_period)
+        self.assertEqual(calculate_last_period(sunday), expected_period)
```
Comment (gcapalbo) [inline custom/ewsghana/tests/test_calculate_last_period.py]: :+1: nice, clear test

File: custom/ewsghana/utils.py [hunk 3]
```diff
@@ -51,19 +51,12 @@ def make_url(report_class, domain, string_params, args):
...
-    if i < 0:
-        # today is Monday, Tuesday or Wednesday -> calculate Thursday from previous week
-        last_th = enddate + timedelta(days=-i, weeks=-1)
-    else:
-        # today is Thursday, Friday, Saturday or Sunday -> calculate Thursday from this week
-        last_th = enddate - timedelta(days=i)
-    fr_before = last_th - timedelta(days=6)
-    return fr_before, last_th + timedelta(days=1)
+# Calculate last full period (Friday - thursday)
+def calculate_last_period(enddate=datetime.utcnow()):
```
Comment (gcapalbo) [inline custom/ewsghana/utils.py]: Will this automatically be used for all reports? sounded from the ticket that Mike wanted it to be the default for all reports.

Comment (kkrampa) [inline custom/ewsghana/utils.py]: yep, this method has been already used in all reports. I only changed a little bit logic.

**Pull request merged**